contracts/access/controller/jstore.go

```diff
@@ -0,0 +1,102 @@
+package controller
+
+import (
+	"io/ioutil"
+	"os"
+	"sync"
+
+	"go.dedis.ch/dela/core/store"
+	"go.dedis.ch/dela/serde"
+	"go.dedis.ch/dela/serde/json"
+	"golang.org/x/xerrors"
+)
+
+// accessStore defines a simple read/write interface to store the access
+type accessStore interface {
+	store.Writable
+	store.Readable
+}
+
+func newJstore(path string) (accessStore, error) {
+	data := map[string][]byte{}
+
+	ctx := json.NewContext()
+
+	if fileExist(path) {
+		buf, err := ioutil.ReadFile(path)
+		if err != nil {
+			return nil, xerrors.Errorf("failed to read file '%s': %v", path, err)
+		}
+
+		err = ctx.Unmarshal(buf, &data)
+		if err != nil {
+			return nil, xerrors.Errorf("failed to read json: %v", err)
+		}
+	}
```

Comment:
There's a subtle issue with this.
If `fileExist(path)`, we might identify issues at startup (corruption, lacking access rights, etc.
If it doesn't, we'll only find out much later that something's wrong (on a call to `saveFile`, most likely).

Would it make sense to create the file immediately, possibly even save after read, to make sure that all foreseeable problems have been detected at startup? (e.g. no write access, parent folder doesn't exist, etc.)

Reply:
nice catch, makes sense!